app/src/main/java/com/example/sharingang/ui/adapters/MessageAdapter.kt

```diff
@@ -75,8 +78,7 @@ class MessageAdapter(
      * @param newData the incoming data
      */
     fun submitList(newData: List<Chat>) {
-        chats.clear()
-        chats.addAll(newData)
+        chats = newData
```

Comment:
Can't this be a problem if the parameter `newData` is changed afterwards?

Reply:
If the newData changes, it means that there was a change in the messages. If there is a change in the messages, it means that there is a listener that will see that change and update the view. Therefore I don't think it is a problem.

Reply:
Ok great!

Reply:
However I needed to revert some changes as there was some issue with the updates so now it's back at how it was before 😅